tempesta_fw/http_parser.c

```diff
@@ -2251,6 +2312,116 @@ __req_parse_host(TfwHttpReq *req, unsigned char *data, size_t len)
 	return r;
 }
 
+/**
+ * Function for parsing 'Referer' header value. This is a partial
+ * copy from parsing procedure for request start line URI with some 
+ * simplifications according to RFC 7231 chapter 5.5.2.
+ */
+static int
+__req_parse_referer(TfwHttpMsg *hm, unsigned char *data, size_t len)
```

Comment:
It appears that only part of the contents of the 'Referer: ' header field is stored in the TfwStr{} string for this header field. I believe this is the first header field parsed this way. This should be commented and reasons explained.

Reply:
I appreciate the effort for strong input validation, this is what we generally do in Tempesta FW. But this particular header is introduced to the set of the special headers just to block particular and very narrow set of attacks which can not mangle the header value, so no need to introduce more parser complexity to validate the header values. So please treat the header value just as an opaque data, with alphabet validation only as we do this for other headers.

Reply:
Done.